src/parser/mod.rs

```diff
@@ -12315,6 +12310,124 @@ impl<'a> Parser<'a> {
         }
         false
     }
+
+    /// Look for all of the expected keywords in sequence, without consuming them
+    fn peek_keywords(&mut self, expected: &[Keyword]) -> bool {
+        let index = self.index;
+        for kw in expected {
+            if !self.parse_keyword(*kw) {
+                self.index = index;
+                return false;
+            }
+        }
+        self.index = index;
+        true
```

Comment:
```suggestion
        let matched = self.parse_keywords(expected);
        self.index = index;
        matched
```
thinking we could simplify this by reusing the existing parse_keywords functionality?

Also can we move this method to the same area of the code as we have the other `peek/parse_keyword*`? so that its easier to find when folks look for that type of functionality